maintainers/scripts/pluginupdate.py

```diff
@@ -395,59 +426,97 @@ def rewrite_input(self, *args, **kwargs):
             )
         )
-        parser.add_argument(
-            "--add",
-            dest="add_plugins",
-            default=[],
-            action="append",
-            help=f"Plugin to add to {self.attr_path} from Github in the form owner/repo",
+        # TODO rename  to "common parser"
+        common = argparse.ArgumentParser(
+            add_help=False,
+            description=(f"""
+                Updates nix derivations for {self.name} plugins.\n
+                By default from {self.default_in} to {self.default_out}"""
+            )
         )
-        parser.add_argument(
+        common.add_argument(
             "--input-names",
             "-i",
             dest="input_file",
             default=self.default_in,
             help="A list of plugins in the form owner/repo",
         )
-        parser.add_argument(
+        common.add_argument(
             "--out",
             "-o",
             dest="outfile",
             default=self.default_out,
             help="Filename to save generated nix code",
         )
-        parser.add_argument(
+        common.add_argument(
             "--proc",
             "-p",
             dest="proc",
             type=int,
             default=30,
             help="Number of concurrent processes to spawn. Setting --github-token allows higher values.",
         )
-        parser.add_argument(
+        common.add_argument(
             "--github-token",
             "-t",
             type=str,
             default=os.getenv("GITHUB_API_TOKEN"),
             help="""Allows to set --proc to higher values.
             Uses GITHUB_API_TOKEN environment variables as the default value.""",
         )
-        parser.add_argument(
+        common.add_argument(
             "--no-commit", "-n", action="store_true", default=False,
             help="Whether to autocommit changes"
         )
-        parser.add_argument(
+        common.add_argument(
             "--debug", "-d", choices=LOG_LEVELS.keys(),
             default=logging.getLevelName(logging.WARN),
             help="Adjust log level"
         )
-        return parser
+
+        subparsers = main.add_subparsers(dest="command", required=True)
+        padd = subparsers.add_parser(
+            "add", parents=[common],
+            description="Add new plugin",
+            add_help=False,
+        )
+        padd.set_defaults(func=self.add)
+        padd.add_argument(
+            "add_plugins",
+            default=None,
+            nargs="+",
+            help=f"Plugin to add to {self.attr_path} from Github in the form owner/repo",
+        )
+
+        # update only a subset ? --all ?
+        pupdate = subparsers.add_parser(
+            "update", parents=[common],
+            # parents= just copies arguments, not the actual help !
+            description="Update all or a subset of existing plugins",
+            add_help=False,
+        )
+        pupdate.set_defaults(func=self.update)
+        return main
```

Comment:
the script is run whenever someone wants to add basic (neo)vim plugin or just want to update the plugins. If treesitter is updated, it updates the grammar to match the lockfile

Reply:
the goal is to make the script more robust, e.g. right now it always need to update the plugins, which sometimes fail. I wish we could just keep the file as is and just add a plugin, hence the PR.

Reply:
I understood that. I was just wondering how frequently someone runs `update` without adding new plugins, because I couldn't find a scheduled workflow for doing so.